### PR TITLE
fix(strategy): handle special case when assigning device 0

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/resource.py [hunk 1]
```diff
@@ -224,6 +224,9 @@ def from_spec(cls, spec: int | str | list[int | str]) -> list[int]:
 
         try:
             if isinstance(spec, int):
+                if spec == 0:
+                    # NOTE: this is a special case for the default GPU placement.
+                    return [0]
                 if spec < -1:
                     raise ValueError
                 return list(range(spec))
```
Comment (larme) [inline src/bentoml/_internal/resource.py:232]: here we actually allow -1. Should we allow -1? For CUDA_VISIBLE_DEVICES we can set to -1 to indicated that we want to disable gpu. But for our case, if users did specify gpu in resource configuration, that means they want to use GPU. So maybe -1 should also be disallowed?

Comment (aarnphm) [reply src/bentoml/_internal/resource.py:232]: with -1, torch will also respect that as well. I think user might get surprised if we don't have the same behaviour

Comment (aarnphm) [reply src/bentoml/_internal/resource.py:232]: I think we should

**Comment by aarnphm** [comment]
closed as for assigning single GPU, it should be `'runners.resource."nvidia.com/gpu"=[0]'`

The -1 behaviour should be addressed in a separate PR @larme

**Pull request closed without merging**